conda_forge_tick/utils.py

```diff
@@ -634,3 +645,225 @@ def extract_requirements(meta_yaml):
+    else:
+        plat_arch = [("win", "64"), ("osx", "64"), ("linux", "64")]
+        for k in set(sub_graph["conda-forge.yml"].get("provider", {})):
+            if "_" in k:
+                plat_arch.append(k.split("_"))
+        varient_yamls = [
+            parse_meta_yaml(meta_yaml, platform=plat, arch=arch)
+            for plat, arch in plat_arch
+        ]
+
+    # this makes certain that we have consistent ordering
+    sorted_varient_yamls = [x for _, x in sorted(zip(plat_arch, varient_yamls))]
+    yaml_dict = ChainDB(*sorted_varient_yamls)
+    if not yaml_dict:
+        logger.error(f"Something odd happened when parsing recipe {name}")
+        sub_graph["bad"] = "make_graph: Could not parse"
+        return sub_graph
+
+    sub_graph["meta_yaml"] = _convert_to_dict(yaml_dict)
+    meta_yaml = sub_graph["meta_yaml"]
+
+    for k, v in zip(plat_arch, varient_yamls):
+        plat_arch_name = "_".join(k)
+        sub_graph[f"{plat_arch_name}_meta_yaml"] = v
+        _, sub_graph[f"{plat_arch_name}_requirements"], _ = extract_requirements(v)
+
+    (
+        sub_graph["total_requirements"],
+        sub_graph["requirements"],
+        sub_graph["strong_exports"],
+    ) = extract_requirements(meta_yaml)
+
+    # handle multi outputs
+    if "outputs" in yaml_dict:
+        sub_graph["outputs_names"] = sorted(
+            list({d.get("name", "") for d in yaml_dict["outputs"]}),
+        )
+
+    # TODO: Write schema for dict
+    # TODO: remove this
+    req = get_requirements(yaml_dict)
+    sub_graph["req"] = req
+
+    keys = [("package", "name"), ("package", "version")]
+    missing_keys = [k[1] for k in keys if k[1] not in yaml_dict.get(k[0], {})]
+    source = yaml_dict.get("source", [])
+    if isinstance(source, collections.abc.Mapping):
+        source = [source]
+    source_keys: Set[str] = set()
+    for s in source:
+        if not sub_graph.get("url"):
+            sub_graph["url"] = s.get("url")
+        source_keys |= s.keys()
+    for k in keys:
+        if k[1] not in missing_keys:
+            sub_graph[k[1]] = yaml_dict[k[0]][k[1]]
+    kl = list(sorted(source_keys & hashlib.algorithms_available, reverse=True))
+    if kl:
+        sub_graph["hash_type"] = kl[0]
+    return sub_graph
+
+
+def load_feedstock(
+    name: str,
+    sub_graph: typing.MutableMapping,
+    meta_yaml: Optional[str] = None,
+    conda_forge_yaml: Optional[str] = None,
+    mark_not_archived: bool = False,
+):
+    """Load a feedstock into subgraph based on its name, if meta_yaml and conda_forge_yaml are provided
+
+    Parameters
+    ----------
+    name : str
+        Name of the feedstock
+    sub_graph : MutableMapping
+        The existing metadata if any
+    meta_yaml : Optional[str]
+        The string meta.yaml, overrides the file in the feedstock if provided
+    conda_forge_yaml : Optional[str]
+        The string conda-forge.yaml, overrides the file in the feedstock if provided
+    mark_not_archived
```

Comment:
This option came from my shims around pulling archived from the admin migrations. We might move a copy of that code here so both things are independent.

Reply:
ok, do you want that in this PR?

Reply:
nope just commenting :)